src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -275,6 +291,39 @@ private long getMeanTimeToRecovery(Run<?, ?> run) {
         return 0;
     }
 
+    private void startTrace(BuildData buildData){
+        if(buildData == null){
+            return;
+        }
+        Tracer tracer = GlobalTracer.get();
+        Span span = tracer.buildSpan("JenkinsBuild").withTag(DDTags.SERVICE_NAME, "jenkins").start();
+        // TODO: Why are both traceId and spanId equal to 0?
+        String traceId = CorrelationIdentifier.getTraceId();
+        String spanId = CorrelationIdentifier.getSpanId();
```

Comment:
These calls get the ID from the current scope.  Instead you should call:
```
span.context().toTraceId()
span.context().toSpanId()
```

(Note: These calls require version 0.32.0 of the OpenTracing API.)